applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -545,15 +639,27 @@ static void handle_dfu_devinfo(uint8_t *data, size_t *size)
 static void handle_reboot_request(uint8_t *data, size_t *size)
 {
 	LOG_INF("System reboot requested");
+	bool is_lock_claimed;
 
-	if (IS_ENABLED(CONFIG_CAF_POWER_MANAGER_EVENTS)) {
-		power_manager_restrict(MODULE_IDX(MODULE), POWER_MANAGER_LEVEL_SUSPENDED);
+	*size = sizeof(bool);
+
+	is_lock_claimed = true;
+	if (IS_ENABLED(CONFIG_DESKTOP_DFU_LOCK)) {
+		is_lock_claimed = dfu_lock_claim(&config_channel_owner);
+		/* No need to release the lock at this point as the system will reboot. */
```

Comment:
You can actually release the lock after calling `sys_reboot` (so that you would not rely on the reboot implementation). Handling such case properly would not hurt (may be better than commenting here actually)